Clipdrop integration

Better-Boy opened the pull request:
## Description

Clipdrop integration

Fixes #8226 

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

ZoranPandovski commented on mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py:
```diff
@@ -0,0 +1,106 @@
+        connection_data = kwargs.get("connection_data", {})
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+        self.client = ClipdropClient(connection_data.get("api_key"), 
```
Maybe move this to the connect method?

Better-Boy replied:
Done

ZoranPandovski commented on mindsdb/integrations/handlers/clipdrop_handler/clipdrop_handler.py:
```diff
@@ -0,0 +1,106 @@
+        # FYI - No way currently to establish a connection with the given key
+        # without making a requests that is chargeable
+        # TODO - Find a way to solve FYI
+        self.is_connected = True
```
Isn't there a healthcheck method that we can call on the client?

Better-Boy replied:
The docs don't mention any healthcheck method.

ZoranPandovski reviewed (changes requested):
Also check the failing test for the pep8 warnings

MinuraPunchihewa reviewed (commented):
@ZoranPandovski Shouldn't this be implemented as a ML handler rather than a API handler?

MinuraPunchihewa reviewed (changes requested):
@Better-Boy Sorry for the trouble, but we believe it is better to implement this as a ML handler rather than an App handler.

Better-Boy commented:
> @Better-Boy Sorry for the trouble, but we believe it is better to implement this as a ML handler rather than an App handler.

@MinuraPunchihewa should I need to convert it to ML handler? If yes, I'll need help from you on https://mindsdbcommunity.slack.com/archives/C04EWFQEFEV/p1698317162200199

Better-Boy commented:
> @Better-Boy Sorry for the trouble, but we believe it is better to implement this as a ML handler rather than an App handler.

@MinuraPunchihewa can clipdrop integration be an API handler for now? Can the conversion to ML handler be taken in another PR?

MinuraPunchihewa commented:
Hey @Better-Boy,
I'm sorry, but I don't think so. Both App and Database handlers are classified as DATA integrations. Since this integration does not really extract/generate any data as such, I don't think it is appropriate to leave this as is.
I will look at your PR and try to resolve the problem you were facing.

Better-Boy commented:
> Hey @Better-Boy, I'm sorry, but I don't think so. Both App and Database handlers are classified as DATA integrations. Since this integration does not really extract/generate any data as such, I don't think it is appropriate to leave this as is. I will look at your PR and try to resolve the problem you were facing.

@MinuraPunchihewa I have converted the clipdrop handler from APIHandler to MLHandler. Please review and let me know if any changes are needed.

MinuraPunchihewa reviewed (changes requested):
@Better-Boy This looks great! If you can address a couple of things, we can go ahead and merge this,
1. Please include a few screenshots or a short video of this integration working as expected.
2. Please add some unit tests for this integration; I have outlined the process for doing so in your StabilityAI PR, but if you have any questions, please feel free to ask me here.

Thanks again for your great work!

Better-Boy commented:
> @Better-Boy This looks great! If you can address a couple of things, we can go ahead and merge this,
> 
> 1. Please include a few screenshots or a short video of this integration working as expected.

I ran it on a VM and deleted the VM. But forgot to record or take screenshot. I'm sure it works. If needed, you can test it. I can provide the API key in slack DM.

> 2. Please add some unit tests for this integration; I have outlined the process for doing so in your StabilityAI PR, but if you have any questions, please feel free to ask me here.
> 
> Thanks again for your great work!

I have added the sample tests.

dusvyat reviewed (approved):
lgtm
